annotations/kubernetes-annotations/src/main/java/io/dekorate/kubernetes/listener/KubernetesSessionListener.java

```diff
@@ -50,20 +50,21 @@
 	public void onClosed() {
     Session session = getSession();
     Project project = getProject();
-    Optional<KubernetesConfig> config = session.configurators().get(KubernetesConfig.class);
-    Optional<ImageConfiguration> imageConfiguration = session.configurators().get(ImageConfiguration.class, BuildServiceFactories.matches(project));
-    if (!config.isPresent()) {
+    Optional<KubernetesConfig> optionalAppConfig = session.configurators().get(KubernetesConfig.class);
+    Optional<ImageConfiguration> optionalImageConfig = session.configurators().get(ImageConfiguration.class, BuildServiceFactories.matches(project));
+    if (!optionalAppConfig.isPresent() || !optionalImageConfig.isPresent()) {
       return;
     }
 
-    KubernetesConfig kubernetesConfig = config.get();
+    KubernetesConfig kubernetesConfig = optionalAppConfig.get();
     Resources resources = session.resources();
     KubernetesList generated = session.getGeneratedResources().getOrDefault(KUBERNETES, new KubernetesList());
 
     BuildService buildService = null;
-    if (kubernetesConfig.isAutoPushEnabled() || kubernetesConfig.isAutoBuildEnabled() || kubernetesConfig.isAutoDeployEnabled()) {
+    ImageConfiguration imageConfig = optionalImageConfig.orElse(null);
+    if (imageConfig.isAutoPushEnabled() || imageConfig.isAutoBuildEnabled() || kubernetesConfig.isAutoDeployEnabled()) {
```

Comment:
Can `imageConfig` be null here?

Reply:
Actually it looks like it can't so maybe just use `optionalImageConfig.get()`?

Reply:
Agree.